pkg/controllers/gateway/gateway_controller.go

```diff
@@ -370,37 +348,36 @@ func (r *GatewayReconciler) reconcileDownstreamFromUpstreamGateway(ctx context.C
 	}
 	//update the cluster set, needs to be ordered or the status update can continually change and cause spurious updates
 	clusters = sets.List(placed)
-	if placed.Equal(targets) {
+	if placed.Equal(targets) && placed.Len() > 0 {
```

Comment:
What does this change do, or rather why is it needed?

Reply:
the programmed status of the gateway was being set to true when the gateway had not been placed. Some of our tests rely on that status, but it was giving a false positive